### DISCUSSION TITLE
Support long metric names in Graph legend in table mode

### DISCUSSION THREAD

**Discussion by nkondratyk** [grafana/grafana#38365] [Dashboards, Panels & Visualizations]
**What happened**:
For some reason I can`t see full output of the metric with Prom labels
![image](https://user-images.githubusercontent.com/34338539/74758298-d2d94500-527f-11ea-84b1-07314ff2c35f.png)

**What you expected to happen**:
I would like to see full output as we can see without legend table. But without table I can`t group by max, min, avg etc..

![image](https://user-images.githubusercontent.com/34338539/74758553-2ba8dd80-5280-11ea-981b-5667c5fbfd65.png)

**How to reproduce it (as minimally and precisely as possible)**:
Create Prom Graph with a lot of labels inside and inside visualization settings use option - "As table"
![image](https://user-images.githubusercontent.com/34338539/74758720-6579e400-5280-11ea-89c6-5b71c31a37f4.png)

**Anything else we need to know?**:
No

**Environment**:
- Grafana version: 6.5.3
- Data source type & version: Prometheus 2.16.0
- OS Grafana is installed on: Ubuntu 16.04
- User OS & Browser: Google Chrome Version 79.0.3945.130
- Grafana plugins:
- Others:

**Comment by torkelo** [comment]
In table mode there is a max width of 650px for each series name. 

Can't you use Legend pattern to only render relevant labels?

**Comment by nkondratyk** [comment]
Got it. Unfortunately I cannot use Legend pattern in some cases, where I don`t know exact list of labels inside.
Is it possible to change max width without touching Grafana source code?

**Comment by torkelo** [comment]
sadly not

**Comment by Mario-Hofstaetter** [comment]
Dear @torkelo , is it possible to manually change this value in a file as a hackaround for the moment?
I searched for `650px`, is this the correct property?
https://github.com/grafana/grafana/blob/81859880d3569fd45a5e2149cbf243c221601f0c/public/sass/components/_panel_graph.scss#L166-L173

I tried changing it but to no success, does this have to be compiled or something? (sorry for the stupid question, I'm no web dev).

**Comment by torkelo** [comment]
Yes , has to be compiled

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by deajan** [comment]
Now, nearly two years later, is this somehow resolved ? I too have truncated legend text, and since the text always begins the same, that doesn't help me in any way.

**Reply by leeoniya** [reply]
i wonder if we should provide a control of where the legend labels ellipsis should occur (left, middle, or right). but also, if there's available screen space, i dont see why we can't provide a way to disable max width or set it explicitly. we'll discuss this soon internally.

cc @adela-almasan @lukasztyrala

**Comment by TimSimmons** [comment]
![Screen Shot 2023-09-11 at 16 06 06](https://github.com/grafana/grafana/assets/5033523/9b1ab2e1-08d2-437d-af41-27adbe69745d)

This also seems to have made it into the explore view in non-table mode. But is there a reason it has to be like this in table mode when there is more room? Could it not overflow with ellipsis at the end of the width of the column in the table view however long that is? Or maybe I'm missing a constraint.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #83015. Feel free to continue the discussion around this there. Thank you!
